boxsdk/object/file.py

```diff
@@ -275,7 +280,14 @@ def update_contents_with_stream(
 
         data = {'attributes': json.dumps(attributes)}
         files = {'file': ('unused', file_stream)}
-        headers = {'If-Match': etag} if etag is not None else None
+        headers = {}
```

Comment:
Did we consider putting this in item.py, so it can be shared between file/folder?

Reply:
Thanks for the suggestion. After looking into this, we feel like it would be better to not consolidate the methods in `item.py`. 

First, to avoid breaking changes, we would have to keep the four methods because they are named differently. It would be nice to consolidate the upload logic since a lot of it is the same, however the  `update_contents_with_stream` in file and `upload_with_stream` in folder do slightly different things. The file method is for uploading a new file version and the folder method is for uploading a new file. Consolidating the logic would require doing some conditional logic in a single method that is part of `item.py` to change between these two calls. This extra logic may just make the SDK a little more confusing.

Reply:
Sorry, I was actually just referring to the new code that we've added to generate the headers, since we've duplicated that in both file/folder. Was thinking something like `_get_headers_for_upload(etag, sha1)`?

Reply:
Ah okay looks like you've discussed this idea too in the other comment below